v2/rest/pulse.go

```diff
@@ -0,0 +1,136 @@
+package rest
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"path"
+	"strconv"
+
+	"github.com/bitfinexcom/bitfinex-api-go/pkg/convert"
+	"github.com/bitfinexcom/bitfinex-api-go/pkg/models/pulse"
+	"github.com/bitfinexcom/bitfinex-api-go/pkg/models/pulseprofile"
+	"github.com/bitfinexcom/bitfinex-api-go/v2"
+)
+
+type PulseService struct {
+	requestFactory
+	Synchronous
+}
+
+// PublicPulseProfile returns details for a specific Pulse profile
+// https://docs.bitfinex.com/reference#rest-public-pulse-profile
+func (ps *PulseService) PublicPulseProfile(nickname string) (*pulseprofile.PulseProfile, error) {
+	if (len(nickname)) == 0 {
+		return nil, fmt.Errorf("nickname is required argument")
+	}
+
+	req := NewRequestWithMethod(path.Join("pulse", "profile", nickname), "GET")
+	raw, err := ps.Request(req)
+	if err != nil {
+		return nil, err
+	}
+
+	pp, err := pulseprofile.NewFromRaw(raw)
+	if err != nil {
+		return nil, err
+	}
+
+	return pp, nil
+}
+
+// PublicPulseHistory returns latest pulse messages. You can specify
+// an end timestamp to view older messages.
+// see https://docs.bitfinex.com/reference#rest-public-pulse-hist
+func (ps *PulseService) PublicPulseHistory(limit, end int) ([]*pulse.Pulse, error) {
+	req := NewRequestWithMethod(path.Join("pulse", "hist"), "GET")
+	req.Params = make(url.Values)
+	req.Params.Add("limit", strconv.Itoa(limit))
+	req.Params.Add("end", strconv.Itoa(end))
+
+	raw, err := ps.Request(req)
+	if err != nil {
+		return nil, err
+	}
+
+	pph, err := pulse.NewFromRaw(raw)
+	if err != nil {
+		return nil, err
+	}
+
+	return pph, nil
+}
+
+// AddPulse submits pulse message
+// see https://docs.bitfinex.com/reference#rest-auth-pulse-add
+func (ps *PulseService) AddPulse(p *pulse.Pulse) (*pulse.Pulse, error) {
+	tl := len(p.Title)
+	if tl < 16 || tl > 120 {
+		return nil, fmt.Errorf("Title length min 16 and max 120 characters. Got:%d", tl)
+	}
+
+	payload, err := json.Marshal(p)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := ps.requestFactory.NewAuthenticatedRequestWithBytes(bitfinex.PermissionWrite, path.Join("pulse", "add"), payload)
+	if err != nil {
+		return nil, err
+	}
+
+	raw, err := ps.Request(req)
+	if err != nil {
+		return nil, err
+	}
+
+	pm, err := pulse.NewSingleFromRaw(raw)
+	if err != nil {
+		return nil, err
+	}
+
+	return pm, nil
+}
+
+// PulseHistory allows you to retrieve your private pulse history or the public pulse history
+// with an additional UID_LIKED field
+// see https://docs.bitfinex.com/reference#rest-auth-pulse-hist
+func (ps *PulseService) PulseHistory(isPublic int) ([]*pulse.Pulse, error) {
```

Comment:
I know that you've copied this from the docs, but lets convert this to a bool. According to the docs `1 if the message is public, 0 if it is not public` - sounds like it is very unlikely that we will need an int in the future so we should be safe to make this assumption